### ISSUE TITLE
Locale warning should probably not be there?

### ISSUE THREAD

**Issue by skanev** [wez/wezterm#2793]
### What Operating System(s) are you seeing this problem on?

macOS

### WezTerm version

wezterm 20221122-154621-68c754b1

### Did you try the latest nightly build to see if the issue is better (or worse!) than your current version?

Yes, and I updated the version box above to show the version of the nightly that I tried

### Describe the bug

When I run wezterm, it fires of this warning:

```
12:38:00.130  WARN   env_bootstrap > setlocale(en_BG.UTF-8) failed, fall back to en_US.UTF-8
```

Here's the relevant code on `main`:

```rust
let lang_code = nsstring_to_str(lang_code_obj);
let country_code = nsstring_to_str(country_code_obj);

let candidate = format!("{}_{}.UTF-8", lang_code, country_code);
let candidate_cstr = std::ffi::CString::new(candidate.as_bytes().clone())
    .expect("make cstr from str");

// If this looks like a working locale then export it to
// the environment so that our child processes inherit it.
let old = libc::setlocale(libc::LC_CTYPE, std::ptr::null());
if !libc::setlocale(libc::LC_CTYPE, candidate_cstr.as_ptr()).is_null() {
    std::env::set_var("LANG", &candidate);
} else {
    log::warn!("setlocale({}) failed, fall back to en_US.UTF-8", candidate);
    std::env::set_var("LANG", "en_US.UTF-8");
}
libc::setlocale(libc::LC_CTYPE, old);
```

Basically, it ends up trying to find `en_BG.UTF-8`, but it doesn't exist (there is no `BG` variant of `en`), so defaulting to `en_US.UTF-8` is perfectly fine.

I know that if I set `LANG` explicitly the warning would go away, but I'd rather avoid figuring out how to set that env when I'm starting WezTerm from the application launcher or Alfred.

I'm wondering whether this warning is necessary? The code does a good job to provide a good fallback if `LANG` is not set, and if it cannot find anything specific, and the user has not set `LANG`, `en_US.UTF-8` is a pretty good default.

Are you happy to remove the warning? If so, I can open a PR to make it easier for you :)

### Configuration

no config
